packages/pages/src/components/analytics/analytics.test.tsx

```diff
@@ -0,0 +1,150 @@
+/**
+ * @jest-environment jsdom
+ */
+import React from "react";
+import { render, screen, fireEvent, waitFor } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+import { TemplateProps } from "../../common/src/template/types";
+import { AnalyticsProvider } from "./provider";
+import { useAnalytics } from "./hooks";
+import { AnalyticsScope } from "./scope";
+
+global.fetch = jest.fn().mockImplementation(
+  jest.fn(() => {
+    return Promise.resolve({ status: 200 });
+  }) as jest.Mock
+);
+
+jest.spyOn(global, "fetch");
+
+const baseProps: TemplateProps = {
+  document: {
+    __: {
+      entityPageSet: {},
+      name: "foo",
+    },
+    uid: 0,
+    businessId: 0,
+    siteId: 0,
+  },
+  __meta: { mode: "development" },
+};
+
+const currentProcess = global.process;
+
+beforeAll(() => {
+  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+  // @ts-ignore
+  global.process = undefined;
+});
+
+beforeEach(() => {
+  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+  // @ts-ignore
+  global.fetch.mockClear();
```

Comment:
I think you want to clear the mock after each instead of before, otherwise another test suite could be affected.

Reply:
I can adjust and see if that works properly. Most of my research on the topic suggested this way.